### ISSUE TITLE
Suggestion - UserKwargsMixins ala django-braces forms mixins

### ISSUE THREAD

**Issue by 7wonders** [pivotal-energy-solutions/django-datatable-view#159]
Personally I need the current user a lot so have added these mixins myself anyway. Thought it might be a nice addition for others. Make sure to add them as the first in the class.

```
class UserDTKwargsMixin(object):
    def get_datatable_kwargs(self, **kwargs):
        kwargs = super(UserDTKwargsMixin, self).get_datatable_kwargs(**kwargs)
        kwargs.update({"user": self.request.user})
        return kwargs


class UserKwargModelDTMixin(object):
    def __init__(self, object_list, url, view=None, callback_target=None, model=None, query_config=None,
                 force_distinct=True, **kwargs):
        self.user = kwargs.pop("user", None)
        super(UserKwargModelDTMixin, self).__init__(object_list, url, view, callback_target, model, query_config,
                                                    force_distinct, **kwargs)
```

Usage:

```
class Table(UserKwargModelDTMixin, Datatable):

class View(UserDTKwargsMixin, DatatableView):
```

**Comment by jangeador** [comment]
Excellent. This is a fine addition. I was about to open an issue asking how to get the user at the DataTable level. As a note for others using it, you need to use the mixin in both the view and the table for it to work. This would definitely be a great addition if @tiliv wants to include it in the code.

**Comment by tiliv** [comment]
Yeah, making the view forward a 'user' kwarg via ``get_datatable_kwargs()`` is the correct way to do it (pretty much like vanilla forms).  The design idea was that the base queryset is all derived from the view, where we the request and user are important for filtering.

I'm open to changing it so we always send the request to the DataTable kwargs.  I think the convenience outweighs the purity of copying the django's design of forms.

Stay tuned!

**Comment by doganmeh** [comment]
I do something similar to my django forms as well :)

**Comment by doganmeh** [comment]
get_datatable_kwargs never gets called in a MultipleDatatableView. Any ideas how to work around it?

**Comment by doganmeh** [comment]
seems I had to override: get_DTNAME_datatable_kwargs instead.
